src/IpcEventDispatcher.cpp

```diff
@@ -0,0 +1,30 @@
+// Copyright 2015, Christopher J. Foster and the other displaz contributors.
+// Use of this code is governed by the BSD-style license found in LICENSE.txt
+
+#include "IpcEventDispatcher.h"
+#include "IpcChannel.h"
+#include "PointViewerMainWindow.h"
+
+
+IpcEventDispatcher::IpcEventDispatcher(QObject* parent, PointViewerMainWindow* getPayload, 
+                                       QByteArray hookSpec, QByteArray hookPayload, int whichShortCut)
+    : QObject(parent),
```

Comment:
Or more appropriately, perhaps just `HookFormatter`?  It's not just the event which is formatted, it's the payload, etc.

Expanding on this, it would make sense to me if we had multiple hook formatter implementations, one for each type of hook payload.  For now I'd say that's an unnecessary extension, but it'd be easy to do later.